src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -158,22 +180,29 @@ end
 
 /-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
 all other coordinates. -/
-def euclidean_space.single [decidable_eq ι] (i : ι) (a : 𝕜) :
+def euclidean_space.single (i : ι) (a : 𝕜) :
   euclidean_space 𝕜 ι :=
-pi.single i a
+by classical; exact pi.single i a
```

Comment:
Is there a reason you've changed this?  It's a bit more work to require explicit decidability, but since you already did the work, I don't see why to undo it.

Reply:
There was a reason, I think relating to something in the `reindex` work, where the linter was getting mad because I had 'decidable_eq ' in the assumptions, but then when I brought it down, I couldn't show that `euclidean_space.single` was what it was supposed to be, because `euclidean_space.single` was built from a `decidable_eq` and the other thing was built from `classical.decidable` or something like that? I don't remember exactly why, and maybe it could be reverted, but my understanding is that it was better to get `[decidable_eq]`s out of the conditions.

Reply:
I got it back in without the classical. Hopefully the linter likes it. It was an issue where I forgot to give decidability conditions on each of the indexing sets for `collected_orthonormal_basis`